autofs-5.1.8 - fix nonstrict fail handling of last offset mount

From: Ian Kent <raven@themaw.net>

When mounting a list of multi-mount offsets the offset mount should
succeed even if there's a mount failure for the non-strict case (the
default).

But currently if the last offset mount fails the multi-mount fails
regardless of whether the mount is non-strict or not.

Signed-off-by: Ian Kent <raven@themaw.net>
---
 CHANGELOG    |    1 +
 lib/mounts.c |    2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/CHANGELOG b/CHANGELOG
index f81b0259..bd1f672c 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -4,6 +4,7 @@
 - improve descriptor open error reporting.
 - fix root offset error handling.
 - fix fix root offset error handling.
+- fix nonstrict fail handling of last offset mount.
 
 19/10/2021 autofs-5.1.8
 - add xdr_exports().
diff --git a/lib/mounts.c b/lib/mounts.c
index 39b7fe81..b4229908 100644
--- a/lib/mounts.c
+++ b/lib/mounts.c
@@ -1940,7 +1940,7 @@ static int tree_mapent_mount_offsets_work(struct tree_node *n, void *ptr)
 			tree_mapent_mount_offsets(oe, !ctxt->strict);
 	}
 
-	return ret;
+	return (ctxt->strict ? ret : 1);
 }
 
 int tree_mapent_mount_offsets(struct mapent *oe, int nonstrict)
